release_bot/utils.py

```diff
@@ -100,10 +100,105 @@ def shell_command(work_directory, cmd, error_message, fail=True):
         cwd=work_directory,
         universal_newlines=True)
+
     configuration.logger.debug(f"{shell.args}\n{shell.stdout}")
     if shell.returncode != 0:
         configuration.logger.error(f"{error_message}\n{shell.stderr}")
         if fail:
             raise ReleaseException(f"{shell.args!r} failed with {error_message!r}")
         return False
     return True
+
+
+def shell_command_get_output(work_directory, cmd):
+    """
+    Same as shell command, but more simple and returns stdout
+    :param work_directory: A directory to execute the command in
+    :param cmd: The shell command
+    :return: stdout of the command
+    """
+    cmd = shlex.split(cmd)
+    shell = subprocess.run(
+        cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        shell=False,
+        cwd=work_directory,
+        universal_newlines=True)
+    success = shell.returncode == 0
+    if not success:
+        return success, shell.stderr
+    return success, shell.stdout
+
+
+def insert_in_changelog(changelog, version, log):
+    """
+    Patches file with new changelog
+    :param changelog: file with changelog
+    :param version: current version
+    :param log: the changelog to insert
+    :return:
+    """
+    content = f"# {version}\n\n{log}\n"
+    try:
+        with open(changelog, 'r+') as file:
+            original = file.read()
+            file.seek(0)
+            file.write(content + original)
+    except FileNotFoundError as exc:
+        configuration.logger.warning(f"No CHANGELOG.md present in repository\n{exc}")
+
+
+def look_for_version_files(repo_directory, new_version):
+    """
+    Walks through repository and looks for suspects that may be hiding the __version__ variable
+    :param repo_directory: repository path
+    :param new_version: version to update to
+    :return: list of changed files
+    """
+    changed = []
+    for root, _, files in os.walk(repo_directory):
+        for file in files:
+            if file in ('setup.py', '__init__.py', 'version.py'):
+                filename = os.path.join(root, file)
+                success = update_version(filename, new_version)
+                if success:
+                    changed.append(filename.replace(repo_directory + '/', '', 1))
+    if len(changed) > 1:
+        configuration.logger.error('Multiple version files found. Aborting version update.')
+    elif not changed:
+        configuration.logger.error('No version files found. Aborting version update.')
+
+    return changed
+
+
+def update_version(file, new_version):
+    """
+    Patches the file with new version
+    :param file: file containing __version__ variable
+    :param new_version: version to update the file with
+    :return: True if file was changed, else False
+    """
+    with open(file, 'r') as input_file:
+        content = input_file.read().splitlines()
+
+    changed = False
+    for index, line in enumerate(content):
+        if line.startswith('__version__'):
```

Comment:
It's a bit more difficult in [setup.py](https://docs.python.org/3/distutils/setupscript.html):
- the key is `'version'`
- the line might start with whitespaces and there might also be a colon at the end

Reply:
It is, however, [PEP 396](https://www.python.org/dev/peps/pep-0396/) states that it's value should be derived from `__version__` variable
>3. When a module (or package) includes a version number, the version SHOULD be available in the `__version__` attribute.


>8. The version attribute in a classic distutils setup.py file, or the PEP 345 [7] Version metadata field SHOULD be derived from the `__version__` field, or vice versa.